### PR TITLE
Convert pipeline configs from JSON to YAML

### LINKED ISSUE #2: Convert recon specs from json to yaml
The JSON format is an unending source of headaches. wdyt about switching to yaml?

### PR DESCRIPTION
**Pull request by tsalo**
Closes #2.

## Changes proposed in this pull request

- Convert JSONs to YAMLs.
- Modify `_load_recon_spec` to support either YAML or JSON files.

### CODE REVIEW

File: qsirecon/data/pipelines/TORTOISE.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+anatomical: []
```
Comment (tsalo) [inline qsirecon/data/pipelines/TORTOISE.yaml:1]: I'd love to take full advantage of the YAML format and add in comments, but that's something someone with actual knowledge should do.

File: docs/building_workflows.rst [hunk 2]
```diff
@@ -182,39 +211,46 @@ for bundle mapping are
...
-        "dipy_dki",
-        "fit_noddi"
-      ]
-    }
-  ]
+.. code-block:: yaml
+
+  nodes:
+  -   action: DKI_reconstruction
+      input: qsirecon
```
[the remark this answers is not in the document]
Comment (mattcieslak) [reply docs/building_workflows.rst:218]: We're going to remove this in all instances except for when the input is another node, right?

It looks like a lot of the other parameters are explicitly specified here - I had left them out of the documentation to keep it simple. If you think it's still readable we could put these configs here in their complete form

File: qsirecon/data/pipelines/TORTOISE.yaml (hunk 1 above)
Comment (mattcieslak) [reply qsirecon/data/pipelines/TORTOISE.yaml:1]: Do you mean like for describing what the parameters do?

Comment (tsalo) [reply qsirecon/data/pipelines/TORTOISE.yaml:1]: I was more thinking about top-level descriptions of the pipelines (e.g., where it comes from, when you'd want to use it), but informal descriptions of the individual parameters would be great too.

File: docs/building_workflows.rst [hunk 3]
```diff
@@ -99,21 +103,43 @@ another node, you can pass them by matching the ``"input"`` field to the
...
+              algorithm: msmt_csd
+              max_sh:
+              - 8
+              - 8
+              - 8
+          mtnormalize: true
+          response:
+              algorithm: dhollander
+      qsirecon_suffix: MRtrix3_act-FAST
+      software: MRTrix3
```
[the remark this answers is not in the document]
Comment (tsalo) [reply docs/building_workflows.rst:123]: I think that should be fine

File: docs/building_workflows.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (tsalo) [reply docs/building_workflows.rst]: My bad, I didn't look too closely at the old pipelines. I just changed all of the examples in this file to match the old ones.

File: qsirecon/data/pipelines/TORTOISE.yaml (hunk 1 above)
Comment (mattcieslak) [reply qsirecon/data/pipelines/TORTOISE.yaml:1]: do we want it in the yaml or the online docs or the docstring? We could always put a link to the online docs for it in the comments. Then those docs would either come from the docs directory or the docstring

Comment (tsalo) [reply qsirecon/data/pipelines/TORTOISE.yaml:1]: Either works for me

**Review by mattcieslak — APPROVED** [review]

**Pull request merged**